Решение финального задания 14 спринта

RecoilSPB opened the pull request.

Recoilvrn committed:
Решение финального задания 14 спринта

z8q commented on src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @GetMapping
+    public List<ItemDto> getItemsByUser(@RequestHeader("X-Sharer-User-Id") @NotNull Long userId) {
```
Можно посмотреть внутри аннотации RequestHeader или здесь, в [доке](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestHeader.html)
про параметр boolean required
```
Defaults to true, leading to an exception being thrown if the header is missing in the request
```
По дефолту выставлен в true,
значит можно не использовать NotNull

z8q commented on src/main/java/ru/practicum/shareit/item/service/ItemService.java:
```diff
@@ -0,0 +1,18 @@
+
+    List<ItemDto> getItemByText(String text);
+
+    ItemDto getItemById(Long itemId, @NotNull Long userId);
```
Будет ли необходимость в NotNull, если контроллер выполнит проверку

z8q commented on src/main/java/ru/practicum/shareit/item/storage/ItemStorageImpl.java:
```diff
@@ -0,0 +1,95 @@
+        if (item.getAvailable() != null) {
+            exsistingItem.setAvailable(item.getAvailable());
+        }
+        log.info("Получен запрос на изменение характеристик вещи");
```
если тут записать в лог сообщение об успешном результате операции,
а существующий лог поставить в начале, то можно будет избежать формирования сообщения с параметром в 76-й строке
то есть могло бы стать так
- "Получен запрос на изменение характеристик вещи"
- "Вещь с id {} не найдена!"
или
- "Получен запрос на изменение характеристик вещи"
- "Характеристики вещи с id {} изменены"

Recoilvrn committed:
1. Переименован метод creatItem на createItem.
2. Убрана аннотация @NotNull для параметров userId, так как @RequestHeader и @PathVariable уже требуют обязательные значения.
3. Добавлена аннотация @NotBlank к параметру text в getItemByText, чтобы исключить пустые строки.

Recoilvrn committed:
Убрана аннотация @NotNull, так как проверка осуществляется на стороне контроллера

Recoilvrn committed:
Убрана аннотация @NotBlank, так как данная проверка не требуется

Recoilvrn committed:
Попоравил метод updateItem

Recoilvrn committed:
1. Переименовал isExist в checkEmailUniqueness и обновил логику, чтобы логирование предупреждало, когда email уже существует.
2. Атомарный счетчик AtomicLong id используется для генерации ID, что улучшает потокобезопасность.
3. Добавлено логирование результатов — сообщения о запросах и успешных результатах (например, при добавлении, обновлении, удалении пользователя).
4. Упрощение метода checkUserAvailability — проверка пользователя сразу с помощью users.get(id), а не отдельной проверки containsKey
